sched/wqueue/kwork_queue.c

```diff
@@ -80,6 +79,7 @@ static void work_timer_expiry(wdparm_t arg)
     }
 
   spin_unlock_irqrestore(&work->wq->lock, flags);
+  sched_unlock();
```

Comment:
do these locks need to be released in this order for some reasons?
otherwise, it's more natural to release them in the reverse order of acquiring them.

Reply:
yes, it need be this order. but all shced_lock like thsi will be removed after https://github.com/apache/nuttx/pull/14578 get  merged.